src/Engine/Graphics/Renderer/IOverlayRenderer.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <Library/Geometry/Size.h>
+
+class IOverlayRenderer {
+ public:
+    virtual ~IOverlayRenderer() = default;
+    virtual void render(const Sizei& outputPresent, int *drawCalls) = 0;
+    virtual void reloadShaders() = 0;
+};
```

Comment:
I have a feeling this should just be a part of `Renderer`. Let's discuss.

Reply:
OK how about this:
1. We drop `IOverlayRenderer`.
2. We keep `NuklearOverlayRenderer`, create an instance of it inside the `OpenGLRenderer`.
3. We add one method to `Renderer` that would render overlays.
4. This method is reimplemented in `NullRenderer` and does nothing.

Reply:
OK looking at the code I now understand why this is not ideal X).